bindsnet/network/nodes.py

```diff
@@ -557,18 +556,18 @@ def step(self, inpts, dt):
             | :code:`inpts` (:code:`torch.Tensor`): Inputs to the layer.
             | :code:`dt` (:code:`float`): Simulation time step.
         '''
-        # Decrement refrac counters.
-        self.refrac_count[self.refrac_count != 0] -= dt
 
         # Check for spiking neurons.
-        self.s = (self.v >= self.thresh) & (self.refrac_count == 0)
+        self.s = (self.v >= self.thresh)
 
         # Refractoriness and voltage reset.
-        self.refrac_count.masked_fill_(self.s, self.refrac)
-        self.v.masked_fill_(self.s, self.reset)
+        self.v = torch.where(self.s, self.c, self.v)
+        self.u = torch.where(self.s, self.u + self.d, self.u)
+
 
         # Apply v and u updates.
-        self.v += dt * (0.04 * (self.v ** 2) + 5 * self.v + 140 - self.u + inpts)
+        self.v += dt * 0.5 * (0.04 * (self.v ** 2) + 5 * self.v + 140 - self.u + inpts)
+        self.v += dt * 0.5 * (0.04 * (self.v ** 2) + 5 * self.v + 140 - self.u + inpts)
```

Comment:
This should only happen once.

Reply:
Our default dt = 1 ms, we getting the same computational instability that Izhikavitch has in the Matlab code

Reply:
I anticipated this; that's why you should use `dt = 0.5`, instead of running the statement twice for all `dt`.

Reply:
dt can only be integers.. otherwise we getting bugs...

Reply:
No, `dt` is a `float`, or it should be, at least.

Reply:
What bugs?